### DISCUSSION TITLE
Dumping threads/messages with slackdump 2.0.1 (or higher) no longer works?

### DISCUSSION THREAD

**Discussion by pican79** [rusq/slackdump#99] [Q&A] [answered]
Hi,
when I try to dump a thread or a message with versions 2.0.1, 2.0.2 or 2.0.3, I get a "not a slack URL" message.
But the message/thread URLs work OK with version 2.0.0.
Is that normal?

My commands look like:
`slackdump.exe https://xxxxxx.slack.com/archives/blabla/p<timestamp>`

**Comment by rusq** [comment]
Hey @pican79 , I'll checked on current master (which is 2.0.3 with some fixes), and it works for me?
```
Slackdump dev Copyright (c) 2018-2077 rusq (build: placeho)

2022/08/03 07:24:50 > checking user cache...
2022/08/03 07:24:50   thread request #    1, fetched:    7, total:        7, process results:  (speed:  26.67/sec, avg:  26.67/sec)
2022/08/03 07:24:50   thread fetch complete, total: 7
2022/08/03 07:24:50 dumped 1 item(s)
2022/08/03 07:24:50 completed, time taken: 800.655733ms
```

Do you mind sharing the exact URL with me, so I could run unit tests against it?  If you could paste the exact url (you can replace the workspace name, that won't affect it) into a text file and upload it on the link below, I'll be able to verify:

https://www.dropbox.com/request/eEK5YK1srbvXPyBmnvys

Most likely it doesn't match the regexp of the thread, and that regexp will need to be adjusted.

**Comment by pican79** [comment]
Hi rusq, thanks for looking into this.
I copied the thread url in a txt file, which I zipped and uploaded to your Dropbox folder (thread_url_discussion99_pican79.zip).
Let me know if you need anything else.

**Reply by rusq** [reply]
Hey @pican79, thanks for your upload - I have included the provided URL in the unit test and it ran fine?  Also, I do not get any error messages when trying to run with this URL (of course it returns channel not found, but there's no parsing error).

Classic "works on my machine" situation 😂

Would you be able to run it with `-trace trace.out` and upload the trace file to the same link, if that's ok?

**Reply by rusq** [reply]
```
Please login in the browser...
2022/08/04 19:08:25 > checking user cache...
2022/08/04 19:08:25   cache expired: it will be recreated.
2022/08/04 19:08:25 error processing: "XKXNXAXXX:15870XXXX.09XX00" (conversation will be skipped): callback error: failed to dump channel:thread XKXNXAXXX:15870XXXX.09XX00: channel_not_found
2022/08/04 19:08:25 dumped 0 item(s)
2022/08/04 19:08:25 completed, time taken: 896.97622ms
```

I suggest to try enclosing the URL on the command line in double quotes ( " ) and see if that works?

**Reply by pican79** [reply]
Enclosing the url in double quotes didn't change anything.
I tried in Cygwin (which I usually use) and in the Win 10 Command Prompt (which I rarely use).

Does it matter if the workspace name (ie what I replaced by xyz in the sample url I uploaded) has this syntax: abc-defgh?

You mentioned a regexp in your first comment. Do you mind showing me where it is located in your source code?

Providing you the trace.out file will take a while because I'd like to "edit" it like [last time](https://github.com/rusq/slackdump/issues/54#issuecomment-1120210051).

**Reply by pican79** [reply]
Actually, I can't even generate a trace file with 2.0.1+. Is it no longer generated in the current working directory?

With 2.0.0, the trace file is generated in the current working directory.

I even tried the 2.0.3 linux binary in my Ubuntu VM.

In your trace sample, am I right in assuming you're using the automatic login method?
My assumption is based on the "Please login in the browser..." line.

I still use the manual method, ie a secrets.txt file containing:
```
SLACK_TOKEN=xoxc-...
COOKIE=xoxd-...
```
Could this explain your not reproducing my problem?

Which OS are you using?

**Reply by rusq** [reply]
I'm using macOS for development and testing, and Windows 10 and ubuntu-linux to test the execution.

The trace file function haven't changed - it should generate it in the current directory.  There are two ways to enable tracing
1. specifying a command line parameter "-trace trace.out"
2. specifying a variable in `secrets.txt`:
      TRACE=trace.out

I wonder if the second variant would work for you?

I haven't had any problems with this - use it all the time - I'll have a look and see if I can reproduce the trace problem on Windows.

**Reply by pican79** [reply]
Adding TRACE=trace.out to my secrets.txt file doesn't generate a trace.out file, even when the command is successful.

Specifying it as a command line parameter works if the "not a slack URL" error does not occur.

Would this error occur too soon in your code and prevent the trace.out file from being generated?

Looking at the regexp you modified, I see that the '=' character is not an allowed.

So I did these tests with version 2.1.1:
slackdump.exe -trace trace.out https://abc-defgh.slack.com/archives/blabla/p<timestamp>
-> the thread url is valid, the command is successful and a trace.out file is generated in the current working directory

slackdump.exe -trace trace.out https://abc=defgh.slack.com/archives/blabla/p<timestamp>
-> the thread url is not valid (and does not even exist), the command returns the "not a slack URL" error and no trace.out file is generated in the current working directory

**Reply by rusq** [reply]
Hey @pican79, I'm glad that 2.1.1 works for you.

My memory is failing me as usual:

1. I gave the wrong instructions the variable name is `TRACE_FILE=`
2. And you're again correct:  the parameters are parsed before the trace is initialised.  The tracing engages after all the parameters were  verified, and if they're invalid - it won't even get to the part of initialising logging or tracing.

Thanks for your tests!

**Comment by rusq** [comment, marked as answer]
@pican79 thanks for your hint about the dash in workspace name - that was it.  I've updated the regexp to allow for dashes (aligned with slack workspace naming requirements), so now it should work for you:

https://github.com/rusq/slackdump/pull/103/files

I'll release 2.1.1 with this fix shortly.

**Reply by rusq** [reply]
https://github.com/rusq/slackdump/releases/tag/v2.1.1

**Reply by pican79** [reply]
Version 2.1.1 fixes my issue. Thanks.
